app/src/main/java/com/example/wangyf/android_dokumi_test/MainActivity.java

```diff
@@ -14,5 +21,28 @@ protected void onCreate(Bundle savedInstanceState) {
 
         TextView textView = (TextView)findViewById(R.id.test1);
         textView.setText("aaaaaaaaaaaaaaaaaaaa");
+
+
+        if (true){
+            Integer i = null;
+            i.toString();
+
+        }
+
+        SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(this).edit();
+        editor.commit();
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wang6793453/android-dokumi-test/blob/a6c0877d903d740a474dba9fb997d65f5693cfac/app/src/main/java/com/example/wangyf/android_dokumi_test/MainActivity.java#L33"></span>[ApplySharedPref] Consider using <code>apply()</code> instead; <code>commit</code> writes its data to persistent storage immediately, whereas <code>apply</code> will handle it in the background<br />
 Consider using <code>apply()</code> instead of <code>commit</code> on shared preferences. Whereas <code>commit</code> blocks and writes its data to persistent storage immediately, <code>apply</code> will handle it in the background.</td>
    </tr>
  </tbody>
</table>